src/ng/httpBackend.js

```diff
@@ -1,7 +1,32 @@
 'use strict';
 
-function createXhr() {
-    return new window.XMLHttpRequest();
+/**
+ * @ngdoc service
+ * @name $xhrFactory
+ *
+ * @description
+ * Factory function used to create XMLHttpRequest objects.
+ *
+ * Replace or decorate this service to create your own custom XMLHttpRequest objects.
```

Comment:
This affects $http and $resource requests alike, right? (since resource uses http). We should probably note this here.